app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CalendarActivity.java

```diff
@@ -483,4 +520,40 @@ protected void onResume(){
         super.onResume();
         refreshWeekView();
     }
+
+    protected void initFilterCheckboxes() {
+        boolean settings = Utils.getSettingBool(this, Const.CALENDAR_FILTER_FIT_SCREEN, false);
+        menuItemFilterFitScreen.setChecked(settings);
+        applyFilterFitScreen(settings);
+
+        settings = Utils.getSettingBool(this, Const.CALENDAR_FILTER_CANCELED, true);
+        menuItemFilterCanceled.setChecked(settings);
+        applyFilterCanceled(settings);
+
+        settings = Utils.getSettingBool(this, Const.CALENDAR_FILTER_8_to_8, false);
+        menuItemFilter8to8.setChecked(settings);
+        applyFilter8to8(settings);
+    }
+
+    protected void applyFilterCanceled(boolean val) {
+        showCanceledEvents = val;
+        onResume();
+    }
+
+    protected void applyFilter8to8(boolean val) {
+        return;
+    }
+
+    protected void applyFilterFitScreen(boolean val) {
+        int hourHeight = 0;
```

Comment:
Please initialize this variable with `defaultHourHeight`. This way you don't need the `else` branch below.

Reply:
@kordianbruck agree, changed